### PR TITLE
Initial TrustGraph integration (backend)

### PR DESCRIPTION
**Pull request by harlantwood**
There is much useful context and short- and long- term discussion of TrustGraph integration in #94.

This PR is backend only, no frontend, to prepare for future full TrustGraph integration.

### Feed
- [x] Feed by weight, over a certain recent time period
  - [x] We query all TrustAtoms created by the Agent
  - [x] We filter for those TrustAtoms above a weight threshold (nominally >= 0)
  - [x] We follow those links to get their targets (followed agents)
  - [x] We fetch posts from those agents which mention the relevant hashtags 
  - [x] We order these posts according to some combination of: weight of tags and recency (don't want to see very old highly weighted at top)

### Following
- [x] *In addition to* our current "follows", we *optionally* add TrustAtoms which can be thought of as "nuanced follow information" -- eg ( "cuisine", 99% ).
  - [x] DNA: create trustatom with topic/weight 
- [x] A default follow is a TrustAtom with no weight (discuss: or 50%) and no topic (discuss: could be `__FOLLOW__` or similar)
- [x] Remove the existing follow links, they are replaced by TrustAtoms

### Unfollowing
- [ ] On "unfollow", we will delete the relevant TrustAtoms 

### Consider
- [ ] TG uses agent_initial_pubkey vs MF uses BOTH agent_latest_pubkey and agent_initial_pubkey

### CODE REVIEW

**Comment by harlantwood** [comment]
The masterful design doc for the "Follow" function in Agent Profile screen:

image

File: dnas/mewsfeed/integrity_zomes/mews/src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,28 @@ pub struct CreateMewInput {
...
+pub struct FollowInput {
+    pub agent: AgentPubKey,
+    pub follow_topics: Vec<FollowTopicInput>,
+    pub follow_other: bool,
+}
+
+#[derive(Debug, Serialize, Deserialize, SerializedBytes)]
+#[serde(rename_all = "camelCase")]
+pub struct FollowTopicInput {
+    pub topic: String,
```
Comment (harlantwood) [inline dnas/mewsfeed/integrity_zomes/mews/src/lib.rs:52]:
```suggestion
    pub topic: Option<String>,
```

File: dnas/mewsfeed/integrity_zomes/mews/src/lib.rs [hunk 2]
```diff
@@ -31,6 +31,28 @@ pub struct CreateMewInput {
...
+    pub now: hdi::prelude::Timestamp,
+    pub oldest_mew_seconds: Option<u64>,
+}
+
+#[derive(Debug, Serialize, Deserialize, SerializedBytes)]
+#[serde(rename_all = "camelCase")]
+pub struct FollowInput {
+    pub agent: AgentPubKey,
+    pub follow_topics: Vec<FollowTopicInput>,
+    pub follow_other: bool,
```
Comment (harlantwood) [inline dnas/mewsfeed/integrity_zomes/mews/src/lib.rs:46]:
```suggestion
```

File: dnas/mewsfeed/coordinator_zomes/mews/src/lib.rs [hunk 3]
```diff
@@ -419,6 +551,23 @@ pub fn get_mews_with_mention(agent_pub_key: AgentPubKey) -> ExternResult<Vec<Fee
     get_mews_from_path(path)
 }
 
+fn get_mews_from_path_by_author(path: Path, author: AgentPubKey) -> ExternResult<Vec<FeedMew>> {
+    let mut full_path = path;
+    full_path.append_component(hdk::hash_path::path::Component::from("by".to_string()));
```
Comment (harlantwood) [inline dnas/mewsfeed/coordinator_zomes/mews/src/lib.rs:556]: nuke the "by"

```suggestion
```

File: README.md [hunk 4]
```diff
@@ -55,10 +55,10 @@ To test out the UI:
 npm start
 ```
 
-To run another agent, open another terminal, and execute again:
+To run 2 agents:
 
 ```bash
-npm start
+npm run network 2
```
Comment (harlantwood) [inline README.md:61]:
```suggestion
npm start
```

File: dnas/mewsfeed/zomes/coordinator/follows/src/lib-trust-atom-funcs.rs [hunk 5]
```diff
@@ -0,0 +1,714 @@
+use hdk::prelude::*;
```
Comment (harlantwood) [inline dnas/mewsfeed/zomes/coordinator/follows/src/lib-trust-atom-funcs.rs:1]: TODO delete this file after harvesting the useful functions

File: dnas/mewsfeed/helpers/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,38 @@
+use hdk::prelude::*;
```
Comment (harlantwood) [inline dnas/mewsfeed/helpers/src/lib.rs:1]: move to top level /crates/helpers
or break out into its own repo
or PR it to 
https://github.com/holochain/hc-utils

File: dnas/mewsfeed/zomes/coordinator/mews/src/all_mews.rs [hunk 7]
```diff
@@ -27,6 +30,116 @@ pub fn get_all_mews_with_context(_: ()) -> ExternResult<Vec<FeedMew>> {
     Ok(feedmews)
 }
 
+#[hdk_extern]
+pub fn get_trusted_mews_with_context(input: RecommendedInput) -> ExternResult<Vec<FeedMew>> {
```
Comment (harlantwood) [inline dnas/mewsfeed/zomes/coordinator/mews/src/all_mews.rs:34]: adapt to get links by hashtag, then FILTER by author (of link!)

**Comment by bierlingm** [comment]
We're gearing up for the first release now and the question is arising: do you think we will be able to launch with it, or should we go ahead without and introduce it after?

We did talk about it being included in the first release in the What's Happening newsletter.

Was great to see you at dweb camp, hope everything's well, @harlantwood !

**Comment by harlantwood** [comment]
Unfortunately I have too many balls in the air, and need to reduce those in order to do a good job with a smaller number.  

So with sadness I’m putting TrustGraph/MewsFeed integration on hold, to be resumed if/when bandwidth is available. 

If anyone in the community wants to pick it up in the meantime I’m happy to advise.

**Pull request closed without merging**